src/allmydata/storage/http_client.py

```diff
@@ -282,3 +279,23 @@ def read_share_chunk(
             raise ClientException(
                 response.code,
             )
+
+    @inlineCallbacks
+    def list_shares(self, storage_index):  # type: (bytes,) -> Deferred[Set[int]]
+        """
+        Return the set of shares for a given storage index.
+        """
+        url = self._client._url(
+            "/v1/immutable/{}/shares".format(_encode_si(storage_index))
+        )
+        response = yield self._client._request(
+            "GET",
+            url,
+        )
+        if response.code == http.OK:
+            body = yield response.content()
+            returnValue(set(loads(body)))
+        else:
+            raise ClientException(
+                response.code,
+            )
```

Comment:
If a server returns a response body with a non-OK status code what happens to it?

Reply:
Hm. I think client gets garbage collected and the underlying connection gets closed when that happens?